xarray/conventions.py

```diff
@@ -621,4 +621,17 @@ def cf_encoder(variables, attributes):
     """
     new_vars = OrderedDict((k, encode_cf_variable(v, name=k))
                            for k, v in variables.items())
+
+    # Remove attrs from bounds variables before encoding
+    # See issue #2921
+    for var in new_vars.values():
+        bounds = var.attrs['bounds'] if 'bounds' in var.attrs else None
+        if bounds and bounds in new_vars:
+            # see http://cfconventions.org/cf-conventions/cf-conventions.html#cell-boundaries # noqa
+            for attr in ['units', 'standard_name', 'axis', 'positive',
+                         'calendar', 'long_name', 'leap_month', 'leap_year',
+                         'month_lengths']:
+                if attr in new_vars[bounds].attrs:
```

Comment:
Should we be a little more careful about which attributes we drop here?
```python
if attr in new_vars[bounds].attrs and attr in var.attrs:
    if new_vars[bounds].attrs[attr] == var[attr]:
        new_vars[bounds].attrs.pop(attr)
```
I'm thinking of the case where we might read in a file that is slightly CF-incompliant in this aspect, e.g. if the `units` encoding were different for the time and time bounds coordinates.  I think in that case if we dropped the `units` attribute from the time bounds coordinate, we would no longer be able to accurately round trip the file.

Reply:
Good idea! I've implemented this now.